src/System.ComponentModel.Composition/src/System/ComponentModel/Composition/Primitives/SerializableCompositionElement.cs

```diff
@@ -0,0 +1,56 @@
+﻿// -----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// -----------------------------------------------------------------------
+using System;
+using Microsoft.Internal;
+
+namespace System.ComponentModel.Composition.Primitives
+{
+    // As most objects that implement ICompositionElement (such as Export, ComposablePart, 
+    // ComposablePartCatalog, etc) are not serializable, this class is used as a serializable 
+    // placeholder for these types when ICompositionElement is used within serializable types,
+    // such as CompositionException, CompositionIssue, etc.
+    internal class SerializableCompositionElement : ICompositionElement
+    {
+        private readonly string _displayName;
+        private readonly ICompositionElement _origin;
+
+        public SerializableCompositionElement(string displayName, ICompositionElement origin)
+        {
+#if FEATURE_SERIALIZATION
```

Comment:
We don't need FEATURE_SERIALIZATION blocks except as needed for serializing exceptions (so, keep any in exceptions, delete the rest, and see if that compiles)
Suggest you do this as a follow up PR as it needs tests also.